caffe2: Fix lint errors in native/int4mm_kernel

EricGriffith opened the pull request:
Summary: See title

Test Plan: Sandcastle

Differential Revision: D72218816




cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168

Skylion007 commented on aten/src/ATen/native/cpu/int4mm_kernel.cpp:
```diff
@@ -1068,6 +1070,7 @@ static void ref_dyn_quant_matmul_4bit_groupwise_kernel(
   };
 
+  // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
   auto lhs_qa8dx_buffer = std::make_unique<int8_t[]>(
       m * (k + sizeof(float) + sizeof(int32_t))); // Allocate for LHS
   int8_t* lhs_qa8dx = lhs_qa8dx_buffer.get();
```
```suggestion
  std::vector<int8_t> lhs_qa8dx_buffer(m * (k + sizeof(float) + sizeof(int32_t)));
  int8_t* lhs_qa8dx = lhs_qa8dx_buffer.data();
```
Wouldn't this be cleaner?

EricGriffith committed:
caffe2: Fix lint errors in native/int4mm_kernel (#150503)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/150503

See title

Test Plan: Sandcastle

Differential Revision: D72218816
